### PR TITLE
[rllib] A3C Configurations

### PR DESCRIPTION
**Pull request by richardliaw**
## What do these changes do?
Modernizes A3C to match optimizer and evaluator framework. This is the first part in partitioning #1339.

### TODO
 - [x] Test functionality

### CODE REVIEW

File: python/ray/rllib/a3c/a3c.py [hunk 1]
```diff
@@ -8,30 +8,50 @@
...
+    # Preprocessing for environment
+    "preprocessing": {
+        # (Image statespace) - Converts image to Channels = 1
+        "grayscale": True,
+        # (Image statespace) - Each pixel
+        "zero_mean": False,
+        # (Image statespace) - Converts image to (dim, dim, C)
+        "dim": 80,
+        # (Image statespace) - Converts image shape to (C, dim, dim)
+        "channel_major": False},
```
Comment (ericl) [inline python/ray/rllib/a3c/a3c.py]: nit: bracket on next line

File: python/ray/rllib/a3c/a3c.py [hunk 2]
```diff
@@ -40,53 +60,25 @@ class A3CAgent(Agent):
...
-        self.agents = [
+        self.local_evaluator = A3CEvaluator(
+            self.env_creator, self.config, self.logdir, start_sampler=False)
+        self.remote_evaluators = [
             RemoteA3CEvaluator.remote(
                 self.env_creator, self.config, self.logdir)
             for i in range(self.config["num_workers"])]
-        self.parameters = self.policy.get_weights()
+        self.optimizer = AsyncOptimizer(
+            self.config, self.local_evaluator, self.remote_evaluators)
```
Comment (ericl) [inline python/ray/rllib/a3c/a3c.py]: Shall we pass in self.config["optimizer"] to be consistent with DQN

File: python/ray/rllib/optimizers/async.py [hunk 3]
```diff
@@ -35,8 +35,7 @@ def step(self):
         # Note: can't use wait: https://github.com/ray-project/ray/issues/1128
         while gradient_queue:
             with self.wait_timer:
-                fut, e = gradient_queue[0]
-                gradient_queue = gradient_queue[1:]
+                fut, e = gradient_queue.pop(0)
```
Comment (ericl) [inline python/ray/rllib/optimizers/async.py]: Isn't this backwards? You want to deque FIFO, not LIFO.

Comment (richardliaw) [reply python/ray/rllib/optimizers/async.py]: I think it's right; if I understand you correctly - 
```
x = [1,2,3]
x.pop(0) == 1
```

Comment (ericl) [reply python/ray/rllib/optimizers/async.py]: Oh ic, missed the zero arg

**Review by ericl — APPROVED** [review]
Lgtm

**Pull request merged**